core/templates/dev/head/pages/exploration-editor-page/services/user-email-preferences.service.spec.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the UserEmailPreferencesService.
+ */
+
+import { UpgradedServices } from 'services/UpgradedServices';
+
+// eslint-disable-next-line max-len
+require('pages/exploration-editor-page/services/user-email-preferences.service.ts');
+
+describe('User Email Preferences Service', function() {
+  var UserEmailPreferencesService = null;
+  var httpBackend = null;
+  var CsrfService = null;
+  var expId = '12345';
+  var sampleResponse = {
+    email_preferences: {
+      mute_feedback_notifications: false,
+      mute_suggestion_notifications: false
+    }
+  };
+
+  beforeEach(angular.mock.module('oppia'));
+
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    var ugs = new UpgradedServices();
+    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
+      $provide.value(key, value);
+    }
+  }));
+
+  beforeEach(function() {
+    angular.mock.module(function($provide) {
+      $provide.value('ExplorationDataService', {
+        explorationId: expId
+      });
+    });
+  });
+
+  beforeEach(angular.mock.inject(function($injector, $q) {
```

Comment:
Is it possible to put all of these `beforeEach` blocks into one?

Reply:
Done! PTAL.